src/main/java/seedu/address/ui/CommandBox.java

```diff
@@ -49,6 +70,66 @@ private void handleCommandEntered() {
         }
     }
 
+    /**
+     * Displays a list of autocomplete entries.
+     */
+    private void setAutocompleteListener() {
+        // Solution below adapted from https://stackoverflow.com/questions/36861056/javafx-textfield-auto-suggestions
```

Comment:
Should this really be here? haha

Reply:
This listener is placed in the `CommandBox` because it is the listener for the `commandTextField` which changes the result of the autocomplete entries whenever there is a change in the text field. 

Also, I need the `handleCommandEntered` method in the `setOnAction` for each `item` in `populatePopup`, so if shifting the whole `setAutocompleteListener` to `AutocompleteManager`, the `AutocompleteManager` will have to take in `CommandBox` as well. This might form a cyclic dependency which I am trying to avoid.

Hmmm, what is your idea and where do you think this should be placed?

Reply:
oh no. I was just referring to the comments.

Reply:
Yup, I think it will be better to put it inside Javadoc.